v1/status_message.go

```diff
@@ -125,8 +125,9 @@ func (m *StatusMessage) HandleApplicationMetadata() error {
 		return err
 	}
 	m.ApplicationMetadataLayerSigPubKey = recoveredKey
-	m.DecryptedPayload = message.Payload
+	// Calculate ID using the wrapped record
 	m.ID = MessageID(m.SigPubKey(), m.DecryptedPayload)
+	m.DecryptedPayload = message.Payload
```

Comment:
It looks like we can calculate the ID in the first line of this method or even in the method handling datasync layer.

Reply:
actually, we can't, because it uses the signature of the message, I have made that more explicit now by using `recoveredKey` directly

Reply:
Ah it was hidden by `m.SigPubKey()` which took a value from `ApplicationMetadataLayerSigPubKey` field in this line. Now it's clear :)